src/storage/src/render/upsert/types.rs

```diff
@@ -723,6 +723,10 @@ where
                 }))
                 .await?;
 
+            self.stats.update_envelope_state_bytes_by(p_stats.size_diff);
+            self.stats
+                .update_envelope_state_count_by(p_stats.values_diff);
+
```

Comment:
I'd rather this be `set`'s and be after the `self.snapshot_stats += stats` at the end. Also, we can update `rehydration_updates` and `rehydration_diff` inline as well, right?

Reply:
I see, still setting cumulative but at every iteration. Yup, the rehydration metrics could be updated thus as well I think. Let me updated.

Reply:
oh i guess this is technically more granular, so we should update `self.snapshot_stats` in this loop

Reply:
Yup, this is at every write. Cool, will update `self.snapshot_stats` instead.

Reply:
Actually this is not more granular, because this block is not the loop but just a peek, moving this after `self.snapshot_stats += stats` might be cleaner and better.

Reply:
Just updated, lmk what you think

Reply:
Btw, there's no `rehydration_diff` though, but a overall total `rehydration_total` which I kept as is (i.e. after completion so that the numbers are accurate)

Reply:
Hmmm...nvm, it can be moved as well. Update incoming 😬

Reply:
Updated and moved after `self.snapshot_stats += stats` instead of completed block.